src/main/java/com/Elessar/app/server/KafkaMsgSender.java

```diff
@@ -0,0 +1,55 @@
+package com.Elessar.app.server;
+
+
+import com.Elessar.proto.P2Pmsg;
+import com.Elessar.proto.P2Pmsg.P2PMsgRequest;
+import com.Elessar.proto.P2Pmsg.P2PMsgResponse;
+import org.apache.kafka.clients.producer.*;
+import org.apache.kafka.common.serialization.LongSerializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * Created by Hans on 3/17/19.
+ */
+public class KafkaMsgSender implements MsgSender {
+    private static final Logger logger = LogManager.getLogger(KafkaMsgSender.class);
+    private final Producer<Long, String> producer;
+
+    public KafkaMsgSender() {
+        Properties props = new Properties();
+        props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
+        props.put(ProducerConfig.CLIENT_ID_CONFIG, "client1");
```

Comment:
I think the client id is used by kafka servers to identify who is sending message to kafka. So probably should use ip address or hostname.

And since the ProducerCreator is kept so maybe we should just use it. Sorry this is different than what I said yesterday. But let's stick to ProducerCreator